NCMP_for_2-1-1_and_2-2-1/WHO_code_test.R

```diff
@@ -0,0 +1,97 @@
+# This is just for testing purposes, raw data should need no cleaning
+lost_data <- paste("Cases:", nrow(clean_NCMP)-nrow(NCMP_data), ",", 
+                   "Percent:", (nrow(NCMP_data)-nrow(clean_NCMP)) / nrow(NCMP_data) * 100) # as above, in %
+lost_data
+
+# regroup imd variable from deciles into quintiles to fit the WHO function requirement
+clean_NCMP$school_imd_quintile <- ceiling(clean_NCMP$schoolindexofmultipledepriv / 2)
+
+if (any(unique(clean_NCMP$school_imd_quintile) %in% c(1:5)) == FALSE){
+  print("ERROR! Check that school_imd_quintile variable ranges from 1 to 5")
+}
+
+
+#main function, takes around 26 mins to run:
+
+#the wealthq argument is fed the regrouped IMD variable and assumes 1 = most deprived, 5 = least deprived
+#typeres argument expects urban/rural breakdown variable, 
+# which is not available in the test data but is reported to be available in the raw data as "PupilUrbanRuralIndicator"
+# schoolgovernmentofficeregion variable is used here as a test for othergr argument (which may take ethnicity from the raw data)
+# reformatting this variable into character, in case it's a factor (to avoid function errors)
+clean_NCMP$schoolgovernmentofficeregion <- as.character(clean_NCMP$schoolgovernmentofficeregion)
+
+test_NCMP <- anthro_prevalence(sex = clean_NCMP$GenderCode,
+                             age = clean_NCMP$AgeInMonths, 
+                             is_age_in_month = TRUE, 
+                             weight = clean_NCMP$Weight, 
+                             lenhei = clean_NCMP$Height, 
+                             measure = "H",
+                             wealthq = clean_NCMP$school_imd_quintile,
+                             othergr = clean_NCMP$schoolgovernmentofficeregion)
+
+#get relevant info only
+rownames(test_NCMP) <- test_NCMP$Group #assigning the first column as rownames (makes is cleaner)
+test_NCMP <- test_NCMP[, -1] #removing the duplicate column
+
+#lots of NA rows for the non-used function arguments, so getting rid of those:
+test_NCMP <- test_NCMP[rowSums(is.na(test_NCMP)) != ncol(test_NCMP), ]
+
+#renaming the default deprivation variable to match dataset:
+first_Q <- which(rownames(test_NCMP)=="Wealth quintile: Q1: Poorest")
+rownames(test_NCMP)[first_Q:(first_Q + 4)] <- c("School IMD Q1: most deprived", 
+                                                "School IMD Q2",
+                                                "School IMD Q3","School IMD Q4",
+                                                "School IMD Q5: least deprived")
+
+
+#these are the columns of interest from the function output (based on the function documentation)
+cols_to_keep <- c("HAZ_pop","HA_2_r", "WH2_r", "WH_2_r") 
+final_result <- test_NCMP[, cols_to_keep] # keeping only the useful columns
+#removing obsolete rows (we only have 1 age group, so deleting repetitions)
+final_result <- final_result[!(row.names(final_result) %in% c("All", "Sex: Female","Sex: Male")), ]
```

Comment:
will this dataset always just include 4+ ages. If they start including under 4s in the future we will need to adapt this code, so maybe just something to make a note on in the readme perhaps?

Reply:
Yes, unlikely to change the age group but will definitely make a note in the readme